v2/yarpcprotobuf/protoc-gen-yarpc-go/internal/template/server_impl.tmpl

```diff
@@ -42,7 +42,20 @@
         if req == nil {
           return nil, yarpcprotobuf.CastError(new({{goType .Request $gopkg}}), m)
         }
-        return h.server.{{.Name}}(ctx, req)
+
+        // This is intentional. While it seems like
+        //
+        //   return h.server.{{.Name}}(ctx, req)
+        //
+        // will behave the same way, the dynamic value of the nil returned is
+        // strongly typed. When passed later as an interface{} inc odec.Encode(...),
```

Comment:
inc odec/in codec

This is a next-level comment. :+1:

This took me a while to understand. Perhaps say specifically the nil response gets carried as a nil-of-type-response when we carry it around as interface {}, so we would prefer to replace it with an absolute nil before we pass it up.